Feature mdx render

enyelsequeira opened the pull request:
Feature is done, please review it and see if needs to be refactored. Now we are able to create pages from MDX and pages render everything as intended `code blocks` within a paragraph as well as `code blocks ` on their own. Numbers are added, only missing highlight if we want that

enyelsequeira committed:
creating pages from mdx-slugs are template started

enyelsequeira committed:
removing unusued code

enyelsequeira committed:
files before sumbmission

enyelsequeira committed:
feature adding mdx-renderer done, need to move root-wrapper to tsx

enyelsequeira committed:
features working correctly

enyelsequeira committed:
adding most needed things styling missing

enyelsequeira commented:
Hey, so most things that you need is there. I haven't started the styling, because I am pretty sure you want me to use the components you have created, so just need a quick run through them so I can understand how you want me to style things. In the meantime. I'll try to see how to implement the filtering.

Kerosz reviewed (changes requested):
There are a couple of things that need adjustments.

1. You still have TypeScript compiler errors, the code compiles and runs but that is only because gatsby by default doesn't run their TypeScript in strict mode, otherwise the code would have not run. In the `blog/card` component the Post interface is not implemented correctly.
![image](https://user-images.githubusercontent.com/31253154/110082148-a2365180-7d95-11eb-9ec6-6ed676c3ff33.png)
In `pages/blog/index.tsx` same issue with `Post` interface
![image](https://user-images.githubusercontent.com/31253154/110082371-ecb7ce00-7d95-11eb-8ed4-0c6cfb90cf00.png)
2. `markdown.tsx` can be moved into it's own component -> `components/blog/markdown/code.tsx`
3. Clean up the files you previously worked on where you have `console.log`s and dead commented code.
4. Remove all dependencies that you no longer need, and all the code from the previous implementation of `contentful-source` that you no longer use from all the files.

**As a guideline anything that goes into the main code, should have no errors, no logs and no dead code**

enyelsequeira committed:
fixing comments

enyelsequeira committed:
fixing comments

enyelsequeira commented:
I've fixed comments and move components around like suggested. Removed old unnecessary code blocks. dependencies wise I didn't remove anything, the main one I was using was  `"@contentful/rich-text-react-renderer": "^14.1.2"` but if I am not mistaken you are also using it in another file as well
